File: works/tmp/5543.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main
{

    public static void main(String[] args) throws IOException
    {
        Scanner sc = new Scanner(System.in);
        int a =  sc.nextInt();
        int b =  sc.nextInt();
        int c =  sc.nextInt();
        int d =  sc.nextInt();
        int e =  sc.nextInt();

        int res1 = Math.min(a, Math.min(b,c));
        int res2 = Math.min(d, e);
        System.out.print(res1+res2-50);
    }


}

File: works/tmp/5543.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.nio.Buffer
import java.util.*


fun main()
{
    val sc = Scanner(System.`in`)
    val a = sc.nextInt()
    val b = sc.nextInt()
    val c = sc.nextInt()
    val d = sc.nextInt()
    val e = sc.nextInt()

    val res1 = Math.min(a, Math.min(b, c))
    val res2 = Math.min(d, e)
    print(res1+res2-50)
}